hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryManager.java

```diff
@@ -128,14 +130,36 @@ void stage(ExpirableTxnRecord record) {
 		txnHistories.computeIfAbsent(txnId, ignore -> new TxnIdRecentHistory()).stage(record);
 	}
 
-	public void purgeExpiredRecordsAt(long now, HederaLedger ledger) {
-		sharedNow = now;
+	public void purgeExpiredRecordsAt(long now) {
+		final var currentAccounts = accounts.get();
 		while (payerExpiries.hasExpiringAt(now)) {
```

Comment:
Although this isn't part of your PR, I'm wondering if you would consider it better for `MonotonicFullQueueExpiries` to contain some method `public Iterator<MerkleEntityId> expiriesAt(long now)` which would replace the use of `hasExpiries(long)` and `expireNextAt(long)`. This would allow `ExpiryManager` to use a for each loop instead of a while loop with a `expireNextAt()` call within it.

Reply:
Interesting idea...will give that some thought. Appreciate you taking the time to look this one over! 🙏